scripts/genpinctrl/config.yaml

```diff
@@ -0,0 +1,158 @@
+---
+pinctrl-addresses:
+    stm32f0: 0x48000000
+    stm32f1: 0x40010800
+    stm32f2: 0x40020000
+    stm32f3: 0x48000000
+    stm32f4: 0x40020000
+    stm32f7: 0x40020000
+    stm32g0: 0x50000000
+    stm32g4: 0x48000000
+    stm32h7: 0x58020000
+    stm32l0: 0x50000000
+    stm32l1: 0x40020000
+    stm32l4: 0x48000000
+    stm32l5: 0x42020000
+    stm32mp1: 0x50002000
+    stm32wb: 0x48000000
+
+alternate-functions:
```

Comment:
`pin-config-templates` would be more in line with what we're intending to do here.